test/src/ui/pages/dashboard/dashboard_notifier_test.dart

```diff
@@ -0,0 +1,91 @@
+import 'package:allthenews/src/domain/communication/api_exception.dart';
+import 'package:allthenews/src/domain/model/article.dart';
+import 'package:allthenews/src/domain/nytimes/ny_times_reactive_repository.dart';
+import 'package:allthenews/src/domain/settings/app_theme.dart';
+import 'package:allthenews/src/domain/settings/popular_news_criterion.dart';
+import 'package:allthenews/src/domain/settings/settings_repository.dart';
+import 'package:allthenews/src/ui/pages/dashboard/dashboard_notifier.dart';
+import 'package:allthenews/src/ui/pages/dashboard/news/popular_news_criterion_message_mapper.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mockito/mockito.dart';
+
+import '../../../../common/change_notifier_test_util.dart';
+import 'news/fake_popular_news_criterion_message_mapper.dart';
+
+class MockNYTimesReactiveRepository extends Mock implements NYTimesReactiveRepository {}
+
+class MockSettingsRepository extends Mock implements SettingsRepository {}
+
+
+
+void main() {
+  DashboardNotifier dashboardNotifier;
+  MockNYTimesReactiveRepository mockNYTimesReactiveRepository;
+  MockSettingsRepository mockSettingsRepository;
+  FakePopularNewsCriterionMessageMapper fakePopularNewsCriterionMessageMapper;
+
+  setUp(() {
+    mockNYTimesReactiveRepository = MockNYTimesReactiveRepository();
+    mockSettingsRepository = MockSettingsRepository();
+    fakePopularNewsCriterionMessageMapper = FakePopularNewsCriterionMessageMapper();
+    dashboardNotifier = DashboardNotifier(
+      mockNYTimesReactiveRepository,
+      mockSettingsRepository,
+      fakePopularNewsCriterionMessageMapper,
+    );
+  });
+
+  group('notifier tests', () {
+    test(
+      'should emit loaded dashboard state when fetching succeed',
```

Comment:
```suggestion
      'should emit loaded dashboard state when fetching articles succeed',
```
? na początku zastanawiałem się co tam pobieramy